### PR TITLE
[Color Picker] Set hex value in text field for custom colors

### PR DESCRIPTION
**Pull request by bjcohen**

### CODE REVIEW

File: src/color_picker.coffee [hunk 1]
```diff
@@ -205,6 +205,14 @@ Ember.Widgets.ColorPicker = Ember.Component.extend
       ])
     ])
 
+  setupCustomColor: (->
+    selectedColor = @get 'selectedColor'
+    for row in @get('colorRows')
+      for color in row
+        return if colorToHex(selectedColor) is color
+    @set 'customColor', @get('selectedColor')
+  ).on 'init'
```
Comment (addepar-andy) [inline src/color_picker.coffee]: this works even without prototype extensions?

Comment (bjcohen) [inline src/color_picker.coffee]: Yeah, that won't unless I wrap it in `Ember.observer`

File: src/color_picker.coffee [hunk 2]
```diff
@@ -205,6 +205,14 @@ Ember.Widgets.ColorPicker = Ember.Component.extend
       ])
     ])
 
+  setupCustomColor: Ember.observer(->
+    selectedColor = @get 'selectedColor'
+    for row in @get('colorRows')
+      for color in row
+        return if colorToHex(selectedColor) is color
```
Comment (igillis) [inline src/color_picker.coffee]: nit: call `colorToHex` outside the loop

File: src/color_picker.coffee [hunk 3]
```diff
@@ -205,6 +205,14 @@ Ember.Widgets.ColorPicker = Ember.Component.extend
       ])
     ])
 
+  setupCustomColor: Ember.observer(->
+    selectedColor = @get 'selectedColor'
+    for row in @get('colorRows')
```
Comment (igillis) [inline src/color_picker.coffee]: nit: can you do `return if @get('colorRows').find (row) -> selectedColor in row`

**Comment by igillis** [comment]
lgtmbc

File: src/color_picker.coffee [hunk 4]
```diff
@@ -205,9 +212,16 @@ Ember.Widgets.ColorPicker = Ember.Component.extend
...
+    selectedColor = colorToHex(@get('selectedColor'))
+    return if @get('colorRows').find (row) ->
+      selectedColor in row.invoke 'toLowerCase'
+    @set 'customColor', @get('selectedColor')
+  ).on 'init'
+
   setCustomColor: Ember.observer ->
     if @get('isCustomColorValid') is true
-      @set 'selectedColor', @get 'customColor'
+      @set 'selectedColor', expandHexColor(@get('customColor'))
```
Comment (bjcohen) [inline src/color_picker.coffee]: Added some stuff here to deal with 3-digit hex codes. Basically, I think `selectedColor` should *always* be a valid, 6-digit code while `customColor` can be whatever the user inputs.

Comment (igillis) [inline src/color_picker.coffee]: agreed

**Pull request merged**